src/main/java/org/tuckey/web/filters/urlrewrite/RequestProxy.java

```diff
@@ -279,6 +288,22 @@ private static HttpRequestBase setupProxyRequest(final HttpServletRequest hsRequ
             }
         }
 
+        // Generate proxy headers
+        if (addProxyHeaders) {
+            String xForwardedForHeader = X_FORWARDED_FOR;
+            String xForwardedForValue = hsRequest.getHeader(xForwardedForHeader);
+            if (xForwardedForValue == null) {
+                xForwardedForValue = hsRequest.getRemoteAddr();
+            } else {
+                final String clientAddr = xForwardedForValue.split(",\\s*")[0].trim();
+                if (!clientAddr.isEmpty()) {
+                    xForwardedForValue += ", " + clientAddr;
```

Comment:
Ok, I am bit puzzled by this part: do we need first address to also be added as last, if I follow code correctly,
foo, bar input will produce foo, bar, foo  output. Is last foo needed? 
As far as I know, first address listed is remote client address (foo in above case)

Reply:
Sorry, I bamboozled this part with some last-minute refactoring yesterday. The expected X-Forwarded-For format is "<Client IP>, <Proxy 1 IP>, <Proxy 2 IP>, etc.", which was definitely not what I had here. This should be fixed as of fc7ff7572fd27c9f6e531d3b13fb22f3a8cade77.